src/basic-samples/GuaranteedSubscriber.js

```diff
@@ -0,0 +1,254 @@
+                ' <protocol://host[:port]> <client-username>@<message-vpn> <client-password>.\n' +
+                'Available protocols are ws://, wss://, http://, https://, tcp://, tcps://');
+            process.exit();
+        }
+        var hosturl = argv.slice(2)[0];
+        subscriber.log('Connecting to Solace message router using url: ' + hosturl);
+        var usernamevpn = argv.slice(3)[0];
+        var username = usernamevpn.split('@')[0];
+        subscriber.log('Client username: ' + username);
+        var vpn = usernamevpn.split('@')[1];
+        subscriber.log('Solace message router VPN name: ' + vpn);
+        var pass = argv.slice(4)[0];
+
+        // create session
+        try {
+            subscriber.session = solace.SolclientFactory.createSession({
+                // solace.SessionProperties
+                url:      hosturl,
+                vpnName:  vpn,
+                userName: username,
+                password: pass,
+            });
+        } catch (error) {
+            subscriber.log(error.toString());
+        }
+
+        // define session event listeners
+        subscriber.session.on(solace.SessionEventCode.UP_NOTICE, function (sessionEvent) {
+            subscriber.log('=== Successfully connected and ready to start the message subscriber. ===');
+            subscriber.startConsume();
+        });
+        subscriber.session.on(solace.SessionEventCode.CONNECT_FAILED_ERROR, function (sessionEvent) {
+            subscriber.log('Connection failed to the message router: ' + sessionEvent.infoStr +
+                ' - check correct parameter values and connectivity!');
+        });
+        subscriber.session.on(solace.SessionEventCode.DISCONNECTED, function (sessionEvent) {
+            subscriber.log('Disconnected.');
+            subscriber.consuming = false;
+            if (subscriber.session !== null) {
+                subscriber.session.dispose();
+                subscriber.session = null;
+            }
+        });
+
+        subscriber.connectToSolace();   
+    };
+
+    // Actually connects the session triggered when the iframe has been loaded - see in html code
+    subscriber.connectToSolace = function () {
+        try {
+            subscriber.session.connect();
+        } catch (error) {
+            subscriber.log(error.toString());
+        }
+    };
+
+    // Starts consuming from a queue on Solace message router
+    subscriber.startConsume = function () {
+        if (subscriber.session !== null) {
+            if (subscriber.consuming) {
+                subscriber.log('Already started subscriber for queue "' + subscriber.queueName + '" and ready to receive messages.');
+            } else {
+                subscriber.log('Starting subscriber for queue: ' + subscriber.queueName);
+                try {
+                    // Create a message subscriber
+                    subscriber.messagesubscriber = subscriber.session.createMessageConsumer({
```

Comment:
For consistency's sake, I would propose we change this to `subscriber.messageConsumer` (Note CamelCase)

Reply:
sure, make sense. Done!